maint/tuning/coll/json_gen.sh

```diff
@@ -0,0 +1,27 @@
+#! /bin/sh
+
+cfile="src/mpi/coll/src/json_autogen.c"
+
+cat > $cfile<<EOF
+/* -*- Mode: C; c-basic-offset:4 ; -*- */
+/*
+ *  (C) 2019 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ */
+
+/* DO NOT EDIT!  This file is autogenerated from the corresponding
+ * json files */
+
+EOF
+
+for f in `find maint/tuning/coll -name '*.json'` ; do
+    full=`echo $f | sed -e 's_maint/tuning/coll/__g'`
+    prefix=`echo $full | sed -e 's_/.*__g' | tr a-z A-Z`
+    suffix=`echo $full | sed -e 's_.*/__g' -e 's/\./_/g'`
+    varname="${prefix}_Coll_${suffix}"
+    echo "const char $varname[] = {" >> $cfile
+    cat $f | awk '{ printf("%s ", $0); }' | \
+        sed -e "s/ //g" -e "s/\(.\)/'\1',/g" >> $cfile
```

Comment:
I wish there can be a more readable output than this but I guess it is ok as long as we trust it and not read it :)

Reply:
That was the simplest way that I could come up with, but I'm open to cleaner code.

Reply:
It might be possible to quote the whole file as a string literal with proper escapes. It probably doesn't matter as long as we understand where the actual source is. On that note, could we add a comment above each generated string list the exact path to the originating json file?

It's also a bit annoying that the generated c file contains super long lines. Some editors or code browsers may not cope well with long lines. is it possible to split the output into neat multiple lines? Never mind if it is too difficult.

Reply:
No, string literals have a compiler-dependent length limit in C, so we can't use that.  Yes, I can add a comment and split it into multiple lines.